src/validation.cpp

```diff
@@ -943,6 +944,16 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
                 return state.Invalid(ValidationInvalidReason::TX_NOT_STANDARD, error("evm tx failed to get sender info %s", result.reason.c_str()), REJECT_INVALID, "evm-sender-info");
             }
 
+            // Block gas limit
+            auto blockGasLimit = evm_try_get_block_limit(result);
+            if (!result.ok) {
+                blockGasLimit = DEFAULT_EVM_BLOCK_GAS_LIMIT;
+            }
+
+            if (txResult.used_gas > blockGasLimit) {
+                return state.Invalid(ValidationInvalidReason::TX_NOT_STANDARD, error("evm tx over block gas limit. Gas used: %d Limit: %d", txResult.used_gas, blockGasLimit), REJECT_INVALID, "evm-over-limit");
+            }
+
```

Comment:
This check is not required since it would have already been caught in pre-validation pipeline. Used gas will never exceed gas limit specified in the evm tx: https://github.com/DeFiCh/ain/blob/master/lib/ain-evm/src/core.rs#L267-L273.

Reply:
Thanks.